gsi/gridmap_eppn_callout/globus_gridmap_eppn.c

```diff
@@ -615,15 +616,37 @@ globus_gridmap_eppn_callout(
         context, subject, &found_identity, shared_user_cert, shared_user_chain, shared_exp);
     if(result == GLOBUS_SUCCESS)
     {
-        if(desired_identity && strcmp(found_identity, desired_identity) != 0)
+        if(desired_identity == NULL)
         {
-            GLOBUS_GRIDMAP_CALLOUT_ERROR(
-                result,
```

Comment:
@msalle same here: this is an error, but `result` here is `GLOBUS_SUCCESS`...

Reply:
True, but there is no error returned by any function. It's just that `found_identity` doesn't match `desired_identity`.
I think it's basically ignored in case it's GLOBUS_SUCCESS. In your case indeed it should be `rc`, not `result`.

Reply:
Bother, too quick. that should have been `result` everywhere in your changed block, incl. in the lines above, since it's the value returned in the `error` block.

Reply:
fixed

Reply:
No, I meant *also* the ones in lines 623 and 624. You're doing a `globus_gss_assist_gridmap` and get a `globus_result_t` back. So you can put that in `result`.
In the else clause, things are different, since there `result` is already an error, and you want (probably) the first error to be returned (Personally I wonder whether they shouldn't first log the old error, and then the new).
In your case, the original result was `GLOBUS_SUCCESS`, which has now become an error.

Reply:
@msalle i pushed a new commit so that the gridmap lookup failures in both cases are the result

Reply:
Looks fine to me!